mibidata/segmentation.py

```diff
@@ -75,11 +86,59 @@ def _quadrant_mean(inds, image):
     upper_left = vals[(inds[0] <= y_center) & (inds[1] < x_center)].sum(axis=0)
     upper_right = vals[(inds[0] < y_center) & (inds[1] >= x_center)].sum(axis=0)
     lower_left = vals[(inds[0] > y_center) & (inds[1] <= x_center)].sum(axis=0)
-    lower_right = vals[(inds[0] >= y_center) & (inds[1] > x_center)].sum(axis=0)
+    lower_right = vals[((inds[0] >= y_center) & (inds[1] > x_center))
+                       | ((inds[0] == y_center) & (inds[1] == x_center))] \
+                       .sum(axis=0)
     quads = np.stack((upper_left, upper_right, lower_left, lower_right), axis=1)
+    # sanity check: test that all pixels in the cell are used
+    assert_array_equal(vals.sum(axis=0), quads.sum(axis=1))
     return np.power(np.product(quads, axis=1), 1 / 4)
 
 
+def _circular_sectors_mean(inds, image, n_sec=8):
+    """Divide a region in circular sectors and get the geometric mean across the
+    sectors.
+
+    Args:
+        inds: A tuple of 2 arrays of the y- and x- indices of the pixels in a
+            segmented region of an image.
+        image: A MibiImage in which the corresponding pixel indices are located.
+        n_sec: number of circular sectors to use. Optional, default is 8.
+
+    Returns:
+        An array whose length is equal to the number of channels in the image.
+        Each value in the array is the geometric mean of the image's integrated
+        channel intensities over the regions's n_sec circular sectors.
+    """
+    # calculate the geometric center of the cells and get the counts
+    y_center, x_center = np.mean(inds, axis=1)
+    vals = image.data[inds]  # has shape (num_pixels_in_cell, num_channels)
+
+    # convert to polar coordinates: y, x -> phi, r
+    inds_pol = np.zeros_like(inds)
+    inds_pol[1], inds_pol[0] = util.car2pol(inds[1], inds[0],
+                                            x_center, y_center)
+    # sanity check: polar coordinate values
+    assert(inds_pol[1] >= 0).all()
+    assert(inds_pol[0] >= 0).all()
+    assert(inds_pol[0] < 360).all()
```

Comment:
Are these necessary in the final version?

Reply:
done